lib/src/rules/unnecessary_parenthesis.dart

```diff
@@ -81,4 +87,24 @@ class _Visitor extends SimpleAstVisitor<void> {
       return;
     }
   }
+
+  /// Returns whether [node] "starts" with whitespace.
+  ///
+  /// That is, is there definitely whitespace after the first token in [node]?
+  bool _expressionStartsWithWhitespace(Expression node) =>
+      // As in, `!(await foo)`.
+      node is AwaitExpression ||
+      // As in, `!(new Foo())`.
+      (node is InstanceCreationExpression && node.keyword != null) ||
```

Comment:
Do we also want to allow parens if there is more than one argument, such as `Foo(a, b)` or even if there is at least one argument `Foo(a + b)`? (Saying "no" and waiting to see whether we get feedback is a fine answer.)

Reply:
I don't think it's confusing when there is whitespace inside _other_, _inner_ parens, such as in a function call. `-(Foo(a + b))` is not more clear than `-Foo(a + b)`.